File: UI/wwwroot/Model/Estudiante_clases.js
//@ts-check
// @ts-ignore
import { EntityClass } from '../WDevCore/WModules/EntityClass.js';
import { Calificaciones } from './Calificaciones.js';
import { Clases } from './Clases.js';
import { Estudiantes } from './Estudiantes.js';
import { Periodo_lectivos } from './Periodo_lectivos.js';
import { Secciones } from './Secciones.js';
class Estudiante_clases extends EntityClass {
    /** @param {Partial<Estudiante_clases>} [props] */
    constructor(props) {
        super(props, 'EntityDbo');
        for (const prop in props) {
            this[prop] = props[prop];
        };
    }
    /**@type {Number}*/ Id;
    /**@type {Date}*/ Transferido;
    /**@type {Date}*/ Retirado;
    /**@type {String}*/ Observaciones;
    /**@type {String}*/ ObservacionesPuntaje;
    /**@type {String}*/ Descripcion;
    /**@type {Date}*/ Created_at;
    /**@type {Date}*/ Updated_at;
    /**@type {Number}*/ Promedio;
    /**@type {Boolean}*/ Repitente;
    /**@type {Number}*/ Reprobadas;
    /**@type {Number}*/ Seccion_id;
    /**@type {Number}*/ Clase_id;
    /**@type {Number}*/ Estudiante_id;
    /**@type {Clases} ManyToOne*/ Clases;
    /**@type {Estudiantes} ManyToOne*/ Estudiantes;
    /**@type {Periodo_lectivos} ManyToOne*/ Periodo_lectivos;
    /**@type {Secciones} ManyToOne*/ Secciones;
    /**@type {Array<Calificaciones>} OneToMany*/ Calificaciones;

    /**
    * @returns {Promise< import('../WDevCore/WModules/CommonModel.js').ResponseServices>}
    */
    async ExportClaseBoletin() {
        return await this.Post("Exporter/ExportClaseBoletin", this);
    }
}
export { Estudiante_clases };

File: AppGenerateFiles/dbo/Model/Estudiante_clases.cs
using CAPA_DATOS;
using System;
using System.Collections.Generic;
using System.Linq;
using System.Text;
using System.Threading.Tasks;
namespace DataBaseModel {
   public class Estudiante_clases : EntityClass {
       [PrimaryKey(Identity = false)]
       public int? Id { get; set; }
       public int? Estudiante_id { get; set; }
       public int? Periodo_lectivo_id { get; set; }
       public int? Clase_id { get; set; }
       public int? Seccion_id { get; set; }
       public DateTime? Transferido { get; set; }
       public DateTime? Retirado { get; set; }
       public string? Observaciones { get; set; }
       public DateTime? Created_at { get; set; }
       public DateTime? Updated_at { get; set; }
       public Double? Promedio { get; set; }
       public bool? Repitente { get; set; }
       public int? Reprobadas { get; set; }
       [ManyToOne(TableName = "Clases", KeyColumn = "Id", ForeignKeyColumn = "Clase_id")]
       public Clases? Clases { get; set; }
       [ManyToOne(TableName = "Estudiantes", KeyColumn = "Id", ForeignKeyColumn = "Estudiante_id")]
       public Estudiantes? Estudiantes { get; set; }
       [ManyToOne(TableName = "Periodo_lectivos", KeyColumn = "Id", ForeignKeyColumn = "Periodo_lectivo_id")]
       public Periodo_lectivos? Periodo_lectivos { get; set; }
       [ManyToOne(TableName = "Secciones", KeyColumn = "Id", ForeignKeyColumn = "Seccion_id")]
       public Secciones? Secciones { get; set; }
       [OneToMany(TableName = "Calificaciones", KeyColumn = "Id", ForeignKeyColumn = "Estudiante_clase_id")]
       public List<Calificaciones>? Calificaciones { get; set; }
   }
}
